### PR TITLE
TIP-1098: Object storage for archives (invalid and input)

### PR DESCRIPTION
**Pull request by BitOne**
**Description (for Contributor and Core Developer)**
This PR configures the archivist flysystem to use an object storage on the CI.
And fixes input file name for uploaded file for import.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php [hunk 1]
```diff
@@ -125,6 +127,26 @@ public function archive(JobExecution $jobExecution)
...
+        if (is_readable($localFilePath)) {
+            $localStream = fopen($localFilePath, 'r');
+
+            $this->filesystem->writeStream($fileKey, $localStream);
+
+            if (is_resource($localStream)) {
+                fclose($localStream);
+            }
+        } else {
+            $this->filesystem->put($fileKey, '');
```
[the remark this answers is not in the document]
Comment (phaseinducer) [reply src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php]: I don't understand the purpose of this line.

File: src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php [hunk 2]
```diff
@@ -125,6 +127,26 @@ public function archive(JobExecution $jobExecution)
         }
 
         $this->writer->flush();
+
+        $fileKey = strtr(
```
Comment (phaseinducer) [inline src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php]: The method is already 40 lines long, does it make sense to extract this new part in another one ?

File: tests/legacy/features/Behat/Context/JobContext.php [hunk 3]
```diff
@@ -294,6 +283,33 @@ protected function getJobInstanceArchives($code)
         return $archives;
     }
 
+    /**
+     * Load the archive in memory and return its in-memory path
```
Comment (phaseinducer) [inline tests/legacy/features/Behat/Context/JobContext.php]: In-memory path ? Looks like it's copied into a temp file.

File: src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (BitOne) [reply src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php]: It was like that before, but in the initialization of the writer.
I suppose the idea is to have an empty file when there's no invalid data.
I didn't change the current behavior.

File: tests/legacy/features/Behat/Context/JobContext.php (hunk 3 above)
Comment (BitOne) [reply tests/legacy/features/Behat/Context/JobContext.php]: Good catch, old comment.

File: src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php (hunk 2 above)
Comment (BitOne) [reply src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php]: Yes, definitely.

File: src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (BitOne) [reply src/Akeneo/Tool/Component/Connector/Archiver/AbstractInvalidItemWriter.php]: I checked on the UI: if there's no invalid data, there's no way to download the empty file. I removed this behavior and the behat are ok.

**Pull request merged**